pkgs/build-support/go/module.nix

```
@@ -156,8 +155,12 @@ let
     # error: empty hash requires explicit hash algorithm
     outputHashAlgo = if vendorHash == "" then "sha256" else null;
   }).overrideAttrs overrideModAttrs;
-
-  package = stdenv.mkDerivation (args // {
+in
+  (builtins.removeAttrs prevAttrs [
+    "overrideModAttrs"
+    "vendorSha256"
+    "vendorHash"
```

```suggestion
    "vendorSha256"
    "vendorHash"
```

Now, we pass `vendorHash` directly as a derivation attribute. As for the long-deprecated `vendorSha256`, we should be able to ignore it, as people who haven't migrated to `vendorHash` would get an error. The only thing we need to filter out is `overrideModAttrs`.